helm/templates/daemonset.yaml

```diff
@@ -28,36 +30,40 @@ spec:
             - {{ $k }}.efs.{{ $v.region }}.amazonaws.com
       {{- end }}
     {{- end }}
-    {{- if .Values.imagePullSecrets }}
+    {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
-      {{- range .Values.imagePullSecrets }}
-        - name: {{ . }}
-      {{- end }}
+        {{- toYaml . | nindent 8 }}
     {{- end }}
+      serviceAccountName: {{ template "aws-efs-csi-driver.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- with .Values.nodeSelector }}
       nodeSelector:
-        beta.kubernetes.io/os: linux
-        {{- with .Values.nodeSelector }}
-        {{- . | toYaml | nindent 8 }}
-        {{- end }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
       hostNetwork: true
+      priorityClassName: {{ .Values.priorityClassName }}
+    {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
```

Comment:
I think it is probably the result of another merge but I don't think it was intentional to only include the DNS policy and config if tolerations are specified.

Reply:
Whoa, yes big mistake here, fixing as well. Definitely a merge error.